### PR TITLE
Chore/refactor request headers

### PR DESCRIPTION
**Pull request by samwelkanda**
Adds a dedicated  `RequestHeaders` class to handle headers

Related to https://github.com/microsoft/kiota/issues/3430

closes #158

### CODE REVIEW

File: kiota_abstractions/request_information.py [hunk 1]
```diff
@@ -95,46 +96,25 @@ def url(self, url: Url) -> None:
...
-    def try_add_request_header(self, key: str, value: str) -> bool:
-        """Try to add an header to the request if it's not already set"""
-        if key and value:
-            lowercase_key = key.lower()
-            if lowercase_key in self.headers:
-                return False
-            self.headers[lowercase_key] = {str(value)}
-            return True
-        return False
+    def add_request_headers(self, headers_to_add: RequestHeaders) -> None:
```
Comment (baywet) [inline kiota_abstractions/request_information.py:103]: is there much value in keeping those two vanity methods? Python would be the only language with those.

File: kiota_abstractions/request_headers.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+from typing import Dict, List, Set, Union
+
+
+class RequestHeaders():
+    "Represents a collection of request headers"
+
+    def __init__(self) -> None:
+        self._headers: Dict[str, Set[str]] = {}
+        self._single_value_headers: Set[str] = {
```
Comment (baywet) [inline kiota_abstractions/request_headers.py:11]: could this be static instead? we don't need an instance of those per headers class instance.

File: kiota_abstractions/request_headers.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+from typing import Dict, List, Set, Union
+
+
+class RequestHeaders():
```
Comment (baywet) [inline kiota_abstractions/request_headers.py:6]: Maybe call that class headers collection instead since it can be used for response headers?

Comment (samwelkanda) [reply kiota_abstractions/request_headers.py]: Makes sense. Updated.

File: kiota_abstractions/request_information.py (hunk 1 above)
Comment (samwelkanda) [reply kiota_abstractions/request_information.py]: Not really needed. Removed.

File: kiota_abstractions/request_headers.py (hunk 2 above)
Comment (samwelkanda) [reply kiota_abstractions/request_headers.py]: I'm not sure I follow. I assumed we create `HeadersCollection` instances and track changes per instance. No?

Comment (baywet) [reply kiota_abstractions/request_headers.py]: I was referring to the collection of single valued headers.

**Review by baywet — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**